src/openloco/windows/News/News.cpp

```diff
@@ -227,24 +227,22 @@ namespace openloco::ui::NewsWindow
 
                 case newsItemSubTypes::vehicle:
                 {
-                    auto vehicle = thingmgr::get<openloco::vehicle>(itemId);
-
-                    if (vehicle->tile_x == -1)
+                    things::vehicle::Vehicle train(itemId);
+                    if (train.head->tile_x == -1)
                         break;
 
-                    vehicle = vehicle->next_car()->next_car();
-                    view.thingId = vehicle->id;
-                    vehicle = vehicle->next_car();
+                    view.thingId = train.veh2->id;
 
-                    if (vehicle->type != vehicle_thing_type::vehicle_6)
+                    if (train.cars.size() != 0)
                     {
-                        view.thingId = vehicle->id;
-                        if (vehicle->type == vehicle_thing_type::vehicle_bogie)
+                        view.thingId = train.tail->id;
+
+                        if (train.cars[0].carComponents[0].front->type == VehicleThingType::bogie)
```

Comment:
The type is always going to be a bogie. There is no need to check for this. What the old code did was the following:

If no cars then thingId of veh2
If it has cars thingId of first front bogie // Note this will never be kept as it will always overwrite with the next one
If it has cars thingId of first body